shoplist/src/main/java/com/netguru/codereview/ui/MainFragment.kt

```diff
@@ -31,35 +27,54 @@ class MainFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
 
-        viewModel!!.shopLists.observe(this, { lists ->
-            val progressBar = view.findViewById<ProgressBar>(R.id.message)
-            val latestIcon = view.findViewById<ImageView>(R.id.latest_list_icon)
+        observe()
 
-            val shopLists = lists.map { mapShopList(it.first, it.second) }.also {
-                latestIcon?.load(it.first().iconUrl)
-            }
+        viewModel.getShopList()
+    }
+
+    private fun observe() {
+        viewModel.shopListsLiveData.observe(viewLifecycleOwner) { result ->
```

Comment:
instead of making fragment is the lifecycle owner to controls the observer, using lifecycle owner instead